File: BahamasEngine/BahamasEngine/DataFrames.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BahamasEngine
{
    struct ExecutionDataFrame
    {
        string EventDateTime;
        string Ticker;
        int TradeId;
        int Action;
        int Units;
        OptionContract Contract;
    }

    public struct OptionDataFrame
    {
        public string Ticker;
        public string EventDate;
        public double Bid;
        public int BidSize;
        public double Ask;
        public int Asksize;
        public double Delta;
    }

    struct FuturesDataFrame
    {
        string Ticker;
        string EventDateTime;
        string ContractSymbol;
        string ExpirationDate;
        double Price;
    }

    public struct OHLCVDataFrame
    {
        public string Ticker;
        public string Date;
        public int TimeIndex;
        public double Open;
        public double High;
        public double Low;
        public double Close;
        public int Volume;
    }
}

File: BahamasEngine/data_frames.py
class OptionDataFrame(object):

    def __init__(self, option_id, event_date, event_timeindex, bid, bid_size,
                    ask, ask_size, delta, underlying_price):
        self._option_id = option_id
        self._event_date = event_date
        self._event_timeindex = event_timeindex
        self._bid = bid
        self._bid_size = bid_size
        self._ask = ask
        self._ask_size = ask_size
        self._delta = delta
        self._underlying_price = underlying_price

    @property
    def bid(self):
        return self._bid

    @property
    def ask(self):
        return self._ask

    @property
    def delta(self):
        return self._delta                            